redhat-developer/observability-operator
# Add Alertmanager SMTP support

[MGDSTRM-7467](https://issues.redhat.com/browse/MGDSTRM-7467)

Allows warning alerts to be sent to a specified email addresses via SendGrid. Changes to the `index.json` file are required, an example of which can be found [here](https://github.com/obrienrobert/observability-resources-mk/tree/MGDSTRM-7467).

To test:
- run the operator locally (CRC)
- an SMTP secret needs to be created and populated with valid SMTP credentials (SendGrid preferably). You can create a free SendGrid account, or just ping me and I can give you credentials from my account (probably the easiest thing here)
- point the external config repository to an external repository that contains these changes [here](https://github.com/obrienrobert/observability-resources-mk/tree/MGDSTRM-7467). You can use this branch, as my email is linked to the credentials from the previous step, I will just change the `to:` field to include your email address
- a warning alert needs to be firing. Adding a `repeat_interval: 1m` field to the SMTP configuration will make the alert fire every `1m`. The above branch also covers this, as a critical alert that fires on CRC has been changes to warning for testing these changes
- ensure that the email is received at the specified email addresses, and the information displayed is clear and concise.
